controller/scheduler-v2/scheduler.go

```diff
@@ -191,15 +139,29 @@ func (s *Scheduler) streamFormationEvents() error {
 		return err
 	}
 
+	current := make(chan struct{})
 	go func() {
+		var isCurrent bool
 	outer:
 		for {
 			for formation := range events {
+				// an empty formation indicates we now have the
+				// current list of formations.
 				if formation.App == nil {
-					// sentinel
+					if !isCurrent {
```

Comment:
Why is this necessary for the formation list? The scheduler won't try to stop anything it doesn't have a formation for.